### PR TITLE
[terra-progress-bar] Added valueText Prop for accessibility

### PR DESCRIPTION
**Pull request by MadanKumarGovindaswamy**
### Summary

**What was changed:**

Added additional valueText Prop for accessibility

**Why it was changed:**

This is required for the Fusion Progress Bar which provides a valueText prop for use cases where alphanumeric text is required instead of a numeric progress indicator. 

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10181 

https://github.com/cerner/terra-core/assets/132886427/3b33e93f-0264-45e3-895d-691ce3502d01

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx [hunk 1]
```diff
@@ -25,8 +25,8 @@ const ProgressBarDefault = () => {
 
   return (
     <div>
-      <label aria-live="polite">{`Progress bar: ${val}%`}</label>
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx:28]: We need to keep this example as a reference for using progressbar with Label (as this example was created as per the A11y team input) and update the other examples to use ValueText prop.

File: packages/terra-progress-bar/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Fixed
+  * Added valueText prop to provide context and status update of progress.
```
Comment (supreethmr) [inline packages/terra-progress-bar/CHANGELOG.md:6]:
```suggestion
* Added
  * Added `valueText` prop to provide custom status update of progress.
```

File: packages/terra-progress-bar/src/ProgressBar.jsx [hunk 3]
```diff
@@ -90,9 +94,10 @@ const ProgressBar = ({
         aria-valuemax={100}
         aria-valuemin={0}
         aria-valuenow={normalizedValue}
-        aria-valuetext={!isMac() ? `${normalizedValue}%` : undefined}
+        aria-valuetext={valText}
         tabIndex="-1"
       />
+      <VisuallyHiddenText aria-live="polite" text={valText} />
```
Comment (supreethmr) [inline packages/terra-progress-bar/src/ProgressBar.jsx:100]:
```suggestion
      {valueText  && <VisuallyHiddenText aria-live="polite" text={valText} />}
```
Let's add visual hidden text only when `valueText` prop is specified

**Comment by supreethmr** [comment]
With addition of translation now progress bar requires react-intl as it's Peer dependency. Should be updated in Progress bar Doc site to include react-intl as peer dependency : https://github.com/cerner/terra-core/blob/c6362bf67d05aaa2a82b6ee69063715ac42ff436/packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/About.1.doc.mdx#L33

File: packages/terra-progress-bar/src/ProgressBar.jsx (hunk 3 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-progress-bar/src/ProgressBar.jsx]: Updated.
19176f6285654364b9774ee21c5e682ecead1a5f
Thanks

File: packages/terra-progress-bar/CHANGELOG.md (hunk 2 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-progress-bar/CHANGELOG.md]: Updated.
19176f6285654364b9774ee21c5e682ecead1a5f
Thanks

File: packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx (hunk 1 above)
Comment (MadanKumarGovindaswamy) [reply packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx:28]: yes. have updated all other examples. 
Thanks

Comment (supreethmr) [reply packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx:28]: ValueText Prop is still being used in this example

Comment (MadanKumarGovindaswamy) [reply packages/terra-core-docs/src/terra-dev-site/doc/progress-bar/example/ProgressBarDefault.jsx:28]: Yes @supreethmr .
Screenshot 2024-02-06 at 12 29 12 PM

**Review by supreethmr — APPROVED** [review]

**Comment by rbsree** [comment]
+1, for Accessibility Review.

**Pull request merged**